src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -291,6 +314,7 @@ def _prepare_abstract_distribution(self):
 
 class AlreadyInstalledCandidate(Candidate):
     is_installed = True
+    source_link = None
```

Comment:
This seems unnecessary?

Reply:
It’s similar to the `is_installed` declaration on `_InstallRequirementBackedCandidate`. Not necessary, but helpful for readability.